src/error.rs

```diff
@@ -104,6 +104,9 @@ pub(super) enum User {
     #[cfg(any(feature = "http1", feature = "http2"))]
     #[cfg(feature = "server")]
     UnexpectedHeader,
+    /// User took too long to send headers
+    #[cfg(feature = "http1")]
+    HeaderTimeout,
```

Comment:
The classification of "user" in hyper has so far meant "user of the library", so the programmer writing some code that uses hyper. I wouldn't consider this a "user error", just probably a variant on the main enum.